### ISSUE TITLE
Sharing Workspace folder between Mac and Windows fails with 'No...repository configuration was found.'

### ISSUE THREAD

**Issue by renzska** [microsoft/azure-repos-vscode#297]
I'm getting the following error which I believe is due to my drive mappings (trying to share a single folder of code between my Mac and a Windows VM. I'm running VSCode on my Mac:

> (team) No Team Services or Team Foundation Server repository configuration was found. Ensure you've opened a folder that contains a repository.

I have a Windows VM in Parallels that I've created a Workspace on a mapped drive pointing to a path on my Mac:

```
Path on Mac: /Users/username/Source/Projects

On Windows, I'm mapping Y: to "Macintosh HD"
On Windows, the "Local Folder" path for the workspace is: Y:\Users\username\Source\Projects
```

Is this configuration supported?

Here is my team-extension.log from my Mac when trying to open the root workspace folder:

```
2017-08-02T17:29:38.035Z - info:   [42219] *** FOLDER: /Users/username/Source/Projects ***
2017-08-02T17:29:38.037Z - info:   [42219] VSTSVSCode/1.121.1 (VSCode 1.14.2; Darwin 16.7.0; Node 7.4.0)
2017-08-02T17:29:38.038Z - debug:  [42219] Looking for an External Context at /Users/username/Source/Projects
2017-08-02T17:29:38.038Z - debug:  [42219] No External Context at /Users/username/Source/Projects
2017-08-02T17:29:38.040Z - debug:  [42219] Looking for TFVC repository at /Users/username/Source/Projects
2017-08-02T17:29:38.040Z - debug:  [42219] TFVC Creating Tfvc object with localPath='undefined'
2017-08-02T17:29:38.040Z - debug:  [42219] Using TFS proxy: 
2017-08-02T17:29:38.040Z - debug:  [42219] TFVC Retrieved from settings; localPath='/usr/local/Cellar/tee-clc/14.120.0/bin/tf'
2017-08-02T17:29:38.041Z - debug:  [42219] TFVC Repository created with repositoryRootFolder='/Users/username/Source/Projects'
2017-08-02T17:29:38.041Z - debug:  [42219] TFVC Repository.CheckVersion
2017-08-02T17:29:38.043Z - debug:  [42219] TFVC: tf add -noprompt -?
2017-08-02T17:29:38.047Z - debug:  [42219] TFVC: spawned new process (duration: 3ms)
2017-08-02T17:29:38.053Z - debug:  [42219] TFVC: spawned new process (duration: 6ms)
2017-08-02T17:29:39.587Z - debug:  [42219] TFVC: add exit code: 0 (duration: 1543ms)
2017-08-02T17:29:39.589Z - debug:  [42219] TFVC Minimum required version: 14.0.4
2017-08-02T17:29:39.589Z - debug:  [42219] TFVC (TF.exe, TF.cmd) version: 14.120.0.201706271643
2017-08-02T17:29:39.590Z - debug:  [42219] TFVC Repository.FindWorkspace with localPath='/Users/username/Source/Projects'
2017-08-02T17:29:39.591Z - debug:  [42219] TFVC: tf workfold -noprompt ********
2017-08-02T17:29:39.595Z - debug:  [42219] TFVC: spawned new process (duration: 4ms)
2017-08-02T17:29:39.775Z - debug:  [42219] TFVC: workfold exit code: 100 (duration: 184ms)
2017-08-02T17:29:39.777Z - debug:  [42219] TFVC errors (via stderr): An argument error occurred: The workspace could not be determined from any argument paths or the current working directory.

2017-08-02T17:29:39.777Z - debug:  [42219] TFVC errors (via stdout): Team Explorer Everywhere Command Line Client (version 14.120.0.201706271643)

 workfold command:

  Creates, modifies, or displays information about the mappings between your
  workspace folders and the source control server folders to which they
  correspond.

 Valid option sets:

  workfold [-collection:<url>] [-workspace:<value>] <serverFolder>
  workfold [-map] [-collection:<url>] [-workspace:<value>] <serverFolder>
           <localFolder>
  workfold [-workspace:<value>]
  workfold <localFolder>
  workfold -cloak [-collection:<url>] [-workspace:<value>]
           <serverFolder>|<localFolder>
  workfold -decloak [-collection:<url>] [-workspace:<value>]
           <serverFolder>|<localFolder>
  workfold -unmap [-collection:<url>] [-workspace:<value>]
           <serverFolder>|<localFolder>


2017-08-02T17:29:39.777Z - error:  [42219] Could not find a workspace with mappings (e.g., not a TFVC repository, wrong version of TF is being used). An argument error occurred: The workspace could not be determined from any argument paths or the current working directory.
```

Any help would be appreciated.

**Comment by renzska** [comment]
I just tried to use VSCode from my Windows VM and it is failing as well with the following logs, so the issue may be different than I thought.

```
2017-08-02T17:29:39.777Z - error:  [42219] Could not find a workspace with mappings (e.g., not a TFVC repository, wrong version of TF is being used). An argument error occurred: The workspace could not be determined from any argument paths or the current working directory.
2017-08-02T17:32:44.521Z - info:   [14396] *** FOLDER: y:\Users\u0225413\Source\WESProjects ***
2017-08-02T17:32:44.526Z - info:   [14396] VSTSVSCode/1.121.1 (VSCode 1.14.2; Windows_NT 10.0.15063; Node 7.4.0)
2017-08-02T17:32:44.528Z - debug:  [14396] Looking for an External Context at y:\Users\u0225413\Source\WESProjects
2017-08-02T17:32:44.528Z - debug:  [14396] No External Context at y:\Users\u0225413\Source\WESProjects
2017-08-02T17:32:44.541Z - debug:  [14396] Looking for TFVC repository at y:\Users\u0225413\Source\WESProjects
2017-08-02T17:32:44.541Z - debug:  [14396] TFVC Creating Tfvc object with localPath='undefined'
2017-08-02T17:32:44.542Z - debug:  [14396] Using TFS proxy: 
2017-08-02T17:32:44.543Z - debug:  [14396] TFVC Retrieved from settings; localPath='C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\TF.exe'
2017-08-02T17:32:44.543Z - debug:  [14396] TFVC Repository created with repositoryRootFolder='y:\Users\u0225413\Source\WESProjects'
2017-08-02T17:32:44.550Z - debug:  [14396] TFVC Repository.CheckVersion
2017-08-02T17:32:44.552Z - debug:  [14396] TFVC: tf add -noprompt -?
2017-08-02T17:32:44.560Z - debug:  [14396] TFVC: spawned new process (duration: 6ms)
2017-08-02T17:32:44.616Z - debug:  [14396] TFVC: add exit code: 0 (duration: 63ms)
2017-08-02T17:32:44.618Z - debug:  [14396] TFVC Minimum required version: 14.102.0
2017-08-02T17:32:44.618Z - debug:  [14396] TFVC (TF.exe, TF.cmd) version: 14.102.25619.0
2017-08-02T17:32:44.619Z - debug:  [14396] TFVC Repository.FindWorkspace with localPath='y:\Users\u0225413\Source\WESProjects'
2017-08-02T17:32:44.619Z - debug:  [14396] TFVC: tf workfold -noprompt ********
2017-08-02T17:32:44.623Z - debug:  [14396] TFVC: spawned new process (duration: 3ms)
2017-08-02T17:32:45.323Z - debug:  [14396] TFVC: workfold exit code: 100 (duration: 703ms)
2017-08-02T17:32:45.325Z - debug:  [14396] TFVC errors (via stderr): Unable to determine the source control server.

2017-08-02T17:32:45.327Z - error:  [14396] Could not find a workspace with mappings (e.g., not a TFVC repository, wrong version of TF is being used). Unable to determine the source control server.
```

**Comment by jeffyoung** [comment]
Hi @renzska.  Thanks for trying out the extension.

I can certainly say that I've not tried this scenario with the extension.  That said, if we can get tf.exe/tf to recognize the workspace, I suspect it should work.

The extension is using either TF.exe or the TEE CLC to make calls to the server. On your Mac, you're using the CLC (of course) and on your Windows VM you appear to be using TF.exe.  First, I would run `tf workfold` at the command line on each and pass the folder that you're having trouble with (the root of your workspace).  This will help us figure out what's going sideways.

Realize that whichever tool you used to create the workspace will be the only tool that will recognize the workspace *until* you run `tf workspaces` with the other tool (that doesn't yet recognize that workspace.)  See this [FAQ](https://github.com/Microsoft/vsts-vscode/blob/master/TFVC_README.md#i-have-workspaces-created-with-visual-studio-can-i-use-the-tee-clc-to-work-with-them) and the one following it for more details (make sure to read the *Note*).  In short, both tf.exe and the CLC cache workspace information on the local machine and they do so in *different* folders.  So if you create a workspace with tf.exe (which will recognize workspaces created with Visual Studio, assuming the major version between the two is the same), tf.exe will recognize it but the CLC (installed on the same machine) will not.  You would need to run `tf workspaces` with the CLC (with the proper collection url) so the CLC will become aware of the workspaces in that collection.

**Comment by renzska** [comment]
@jeffyoung Thanks for the feedback.

I was able to get further in the process on the Windows side, but now I'm getting a message about the workspace being a "server" workspace instead of a "local" workspace. It is a local workspace and I believe the issue with this is resolved with this pull request given my logs: https://github.com/Microsoft/vsts-vscode/pull/291

```
2017-08-10T16:39:08.135Z - debug:  [19200] TFVC errors (via stderr): TF30063: You are not authorized to access server\Collection.

2017-08-10T16:39:08.135Z - error:  [19200] Caught an exception during Tfvc SCM Provider initialization
2017-08-10T16:39:08.136Z - error:  [19200] It appears you are using a Server workspace. Currently, TFVC support is limited to Local workspaces. TF30063: You are not authorized to access server\Collection.
```

I'm still getting issues on the Mac side with running the tf workfold command within the folder returning the following:

```
tf workfold
An argument error occurred: The workspace could not be determined from any argument paths or the current working directory.
```
On the Mac side, do I need to specify any commandline parameters when ran from the workspace folder? On the Windows side it worked without a hitch:

```
Y:\Users\username\Source\Work\Projects>tf workfold
===============================================================================
Workspace : Projects (John R)
Collection: http://server:8080/tfs/collection
 $/Projects: Y:\Users\username\Source\Work\Projects
```

Thanks for your help.

**Comment by jeffyoung** [comment]
I merged #291 just a bit ago and should be able to get a release out early next week.

Presumably you created the workspace with either tf.exe or Visual Studio within that Windows VM (although it's pointing to your Mac hard disk).  If that's correct, then tf.exe is seeing the workspace fine since it (or VS IDE) created the workspace.  Since it's local, the information is stored in the cache for tf.exe on the Windows VM.  My guess is that on the Mac side, using the TEE CLC, it isn't aware of the workspace on the hard drive (even though there's only a single set of source in one location) because the workspace wasn't created by the CLC.

On your Mac, try running `tf workspaces -collection:http://server:8080/tfs/collection` with the same values that the `tf workfold` command on Windows returns.  Hopefully, if I understand the scenario correctly, the CLC's cache will be updated with information about the workspaces in that collection and the CLC will be able to map the local folder to the server.  After running workspaces, just try `tf workfold` on the Mac again and see if you get different results.  🤞

**Comment by renzska** [comment]
@jeffyoung,

Thanks for the help, however, I am unable to get it to work. From what I've found, I have to have the same workspace configured on my Mac and my VM, but it seems like that is not a supported scenario unless I update the computername every time I switch the machine I'm using (https://stackoverflow.com/questions/28752591/multiple-computers-sharing-one-tfs-workspace). I was hoping for a simple method to make this work along the lines of git.

I can create two different workspaces (requiring different paths on my system), one for each system, but then that means I have two folders of my source code and I have to keep track of which one is the one I'm currently working on.

My tests:

```
When trying to add an existing Public Workspace it says that it is currently on a different computer:

tf workspace -collection:http://server:8080/tfs/wescollection -new Test

tf workfold -collection:http://server:8080/tfs/wescollection -workspace:Test -map $/TestProjects /Users/username/Source/Work/Test 

tf workfold -collection:http://server:8080/tfs/wescollection -workspace:Test -map $/TestProjects Y:/Users/username/Source/Work/Test 

It errors saying it is already on a different computer. In VS on Windows, it shows as a Remote Workspace with no way to pull it down. When I try to add an additional working folder, it errors: "The path $/TestProjects is already mapped in workspace Test".

At this point, I'm not sure what to try next as it seems I've hit a dead end.

The above commands you mentioned only ever show the respective workspaces that were created on the PC or the Mac, but never across the platforms.

Thanks again for your help trying to get this working.
